### DISCUSSION TITLE
Nextfow skips one of the process

### DISCUSSION THREAD

**Discussion by awast** [nextflow-io/nextflow#4542] [Q&A]
Hi,
  I am running gatk based workflow using nextflow.I have attached the screenshot of running pipeline. It is working fine till panelnormalsone process but skipped the GenomicsDBcreation  process  and jumps to next process (tumorsamplenamerun). 
I have attached the part of the script below.
[gatk-based-script.txt](https://github.com/nextflow-io/nextflow/files/13463245/gatk-based-script.txt)

[0b/0a2657] process > trimming  10 of 10, cached: 10 ✔
[63/45ed0f] process > BWAMEM (6)                    [100%] 10 of 10, cached: 10 ✔
[26/6b75ca] process > gatkrun (10)                  [ 70%] 7 of 10, cached: 7
[93/333915] process > normalsamplenamerun (1)       [100%] 5 of 5, cached: 5
[cc/f08592] process > panelnormalsone (1)           [100%] 5 of 5, cached: 5
[-        ] process > GenomicsDBcreation            -
[fe/bf520d] process > tumorsamplenamerun (2)        [  0%] 0 of 2

Please let me know how can i resolve this problem

**Comment by mribeirodantas** [comment]
There's something wrong with your pipeline script. In the workflow block you call `Genomics_DB_creation` but this process doesn't exist in this file (maybe `GenomicsDBcreation`?). I assume the code you're running is not the code you shared above.

**Comment by awast** [comment]
Thanks for the reply Sir,
      I am running the very same code, actually its a spelling mistake while uploading the script. i have shared the refined code. I have shared the part of the code which causes error along with before and after running process. The output of GenomicsDBcreation process will be used by MM process.
[gatk-based-script.txt](https://github.com/nextflow-io/nextflow/files/13533513/gatk-based-script.txt).
Related to this script if i will run this script till GenomicsDBcreation process then its working fine but if i am running the whole script then this process got skipped.

**Comment by awast** [comment]
hi,
 after doing some changes i have tried the code again, this time it is only working with one sample at a time for example: if i will use only map operator then it is running but taking only sample and  if I am using collect operator also then it skips the process. i have mentioned the code and error below:
like: if i am using **collect** and **map** operator then it skips the process:
panelnormalsone(normalsamplenamerun.out)
  samples = panelnormalsone.out.vcf_normal | map {  t -> t[1] }
  stats = panelnormalsone.out.vcf_stats | map { t -> t[1] }
  tbi = panelnormalsone.out.vcf_tbi | map { t -> t[1] }  
  GenomicsDBcreation( samples.collect(), stats.collect(), tbi.collect() )
[0b/0a2657] process > trimming (trim on {CaGB17_T}) [100%] 10 of 10, cached: 10 ✔
[63/45ed0f] process > BWAMEM (4)                    [100%] 10 of 10, cached: 10 ✔
executor >  local (5)
[0b/0a2657] process > trimming (trim on {CaGB17_T}) [100%] 10 of 10, cached: 10 ✔
executor >  local (5)
[0b/0a2657] process > trimming (trim on {CaGB17_T}) [100%] 10 of 10, cached: 10 ✔
[63/45ed0f] process > BWAMEM (4)                    [100%] 10 of 10, cached: 10 ✔
[81/6812de] process > gatkrun (10)                  [ 70%] 7 of 10, cached: 7
[93/d23613] process > normalsamplenamerun (5)       [100%] 5 of 5, cached: 5
[cc/f08592] process > panelnormalsone (5)           [100%] 5 of 5, cached: 5
[-        ] process > GenomicsDBcreation            -
[6c/1e1ed4] process > tumorsamplenamerun (2)        [  0%] 0 of 2
[-        ] process > MM                            -
[-        ] process > learnReadOrientation          -
[27/903853] process > getpileupnormal (5)           [100%] 5 of 5, cached: 5

but if i am **not** using the **collect** operator then :
samples = panelnormalsone.out.vcf_normal | map {  t -> t[1] }
  stats = panelnormalsone.out.vcf_stats | map { t -> t[1] }
  tbi = panelnormalsone.out.vcf_tbi | map { t -> t[1] }  
  GenomicsDBcreation( samples, stats, tbi)
 its running the process but only for one sample
then,[a9/9d3714] process > trimming (trim on {CaGB28_T}) [100%] 10 of 10, cached: 10 ✔
[63/45ed0f] process > BWAMEM (4)                    [100%] 10 of 10, cached: 10 ✔
[75/805944] process > gatkrun (10)                  [100%] 7 of 7, cached: 7
[93/d23613] process > normalsamplenamerun (3)       [100%] 5 of 5, cached: 5
[23/9379f0] process > panelnormalsone (1)           [100%] 5 of 5, cached: 5
[a4/990c74] process > GenomicsDBcreation (2)        [100%] 1 of 1, failed: 1
[-        ] process > tumorsamplenamerun (2)        -
[-        ] process > MM                            -
[-        ] process > learnReadOrientation          -
[9b/88abc4] process > getpileupnormal (2)           [100%] 5 of 5, cached: 5
 java -Dsamjdk.use_async_io_read_samtools=false -Dsamjdk.use_async_io_write_samtools=true -Dsamjdk.use_async_io_write_tribble=false -Dsamjdk.compression_level=2 -Xmx24G -jar GenomicsDBImport -R path/GCA_000001405.15_GRCh38_no_alt_analysis_set.fna -L path/GB_Exome_Probes-medgenome.interval_list --genomicsdb-workspace-path pondb -V CaGB27_N_recal_somatic.vcf.gz --merge-input-intervals true --reader-threads 10
here it taking only one sample now, but i have include more samples like **-V sample1 -V Sample2  -V Sample3**

Earlier i was getting the same error but your team suggested to use collect operator and then it was working fine but now its not working as i am running the more processes after this process as a part of pipeline.I think the problem is with collect() operator.
It will be great help if you will sugggest me how to make it work
